tests/test_search_utils.py

```diff
@@ -34,6 +34,42 @@ def test_gen_mh():
     return mh.copy_and_clear()
 
 
+def test_add_hashes_at_ranks_0_():
+    lin1 = lca_utils.make_lineage('a;b;c;d;e;f;g')
+    hashval1 = 12345678
+    print(lin1)
+
+    lineage_hashD = defaultdict(test_gen_mh)
+
+    add_hashes_at_ranks(lineage_hashD, [hashval1], lin1, 'genus')
+
+    for rank in 'superkingdom', 'phylum', 'class', 'order', 'family', 'genus':
+        check_lin = pop_to_rank(lin1, rank)
+        assert list(lineage_hashD[check_lin].hashes) == [hashval1]
+
+    # CTB: I believe this should NOT work because lin1 is at species level
```

Comment:
yep. I was popping to rank outside of this fn, but think it's worth popping to rank here too. I made some changes to the `add_hashes_at_rank` function in master that should fix.